ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/metrics/RocksDBMetricsCollector.java

```diff
@@ -257,10 +257,10 @@ private Interval(final int intervalSeconds) {
 
     boolean check() {
       final Instant now = clock.get();
-      return last.accumulateAndGet(
+      return Objects.equals(last.accumulateAndGet(
           now,
           (l, n) -> n.isAfter(l.plusSeconds(intervalSeconds)) ? n : l
-      ) == now;
```

Comment:
hey @rodesai I just threw this fix into my PR when I saw the compiler warning, but now we're suddenly seeing failures in `RocksDBMetricsCollectorTest.shouldNotUpdateIfWithinInterval` due to 
```
org.mockito.exceptions.verification.TooManyActualInvocations: 

kafkaMetric.metricValue();
Wanted 1 time:
-> at org.apache.kafka.common.metrics.KafkaMetric.metricValue(KafkaMetric.java:54)
But was 3 times:
-> at io.confluent.ksql.execution.streams.metrics.RocksDBMetricsCollector$AggregatedMetric.update(RocksDBMetricsCollector.java:229)
-> at io.confluent.ksql.execution.streams.metrics.RocksDBMetricsCollector$AggregatedMetric.update(RocksDBMetricsCollector.java:229)
-> at io.confluent.ksql.execution.streams.metrics.RocksDBMetricsCollector$AggregatedMetric.update(RocksDBMetricsCollector.java:229)
	at io.confluent.ksql.execution.streams.metrics.RocksDBMetricsCollectorTest.shouldNotUpdateIfWithinInterval(RocksDBMetricsCollectorTest.java:256)
```

The thing is, I can't seem to reproduce it locally after thousand+ runs, and it also didn't fail on the build for my PR. Although after looking at the implementation of `accumulateAndGet` I see that it was actually fine before, I still don't understand why this change could be incorrect, much less nondeterministically so. Any thoughts on what's going on here?